### PR TITLE
[SE-0413] Add a throws clause to do..catch statements

### PR DESCRIPTION
**Pull request by DougGregor**
Per the ongoing review of SE-0413, introduce the `do throws(...)` syntax for `do..catch` blocks.

### CODE REVIEW

**Comment by DougGregor** [comment]
Only the last three commits are new; the rest are part of https://github.com/apple/swift-syntax/pull/2379

**Comment by DougGregor** [comment]
https://github.com/apple/swift/pull/70182

@swift-ci please test

**Comment by DougGregor** [comment]
https://github.com/apple/swift/pull/70182

@swift-ci please test Windows

**Comment by DougGregor** [comment]
https://github.com/apple/swift/pull/70182

@swift-ci please test Windows

File: Sources/SwiftSyntax/generated/syntaxNodes/SyntaxNodesD.swift [hunk 1]
```diff
@@ -3495,7 +3503,7 @@ public struct DoStmtSyntax: StmtSyntaxProtocol, SyntaxHashable, _LeafStmtSyntaxN
     }
   }
   
-  public var unexpectedBetweenDoKeywordAndBody: UnexpectedNodesSyntax? {
+  public var unexpectedBetweenDoKeywordAndThrowsClause: UnexpectedNodesSyntax? {
```
Comment (ahoppen) [inline Sources/SwiftSyntax/generated/syntaxNodes/SyntaxNodesD.swift:3506]: Related to https://github.com/apple/swift-syntax/pull/2379#discussion_r1414399964, this needs a compatibility layer or a mention in the release notes.

File: Sources/SwiftSyntaxBuilder/generated/BuildableNodes.swift [hunk 2]
```diff
@@ -381,7 +381,9 @@ extension DoStmtSyntax {
       leadingTrivia: Trivia? = nil, 
       unexpectedBeforeDoKeyword: UnexpectedNodesSyntax? = nil, 
       doKeyword: TokenSyntax = .keyword(.do), 
-      unexpectedBetweenDoKeywordAndBody: UnexpectedNodesSyntax? = nil, 
+      unexpectedBetweenDoKeywordAndThrowsClause: UnexpectedNodesSyntax? = nil, 
+      throwsClause: ThrowsClauseSyntax? = nil, 
+      unexpectedBetweenThrowsClauseAndBody: UnexpectedNodesSyntax? = nil,
```
Comment (ahoppen) [inline Sources/SwiftSyntaxBuilder/generated/BuildableNodes.swift:386]: Also related to https://github.com/apple/swift-syntax/pull/2379#discussion_r1414399964, this needs a compatibility layer or a mention in the release notes.

**Pull request merged**